### ISSUE TITLE
System map not loading properly on server

### ISSUE THREAD

**Issue by KEEGUNATOR2** [Minecraft-Transit-Railway/Minecraft-Transit-Railway#798]
**Expected Behaviour**

The system map to load when doing (server ip):8888/index.html, like when using localhost

**Actual Behaviour**
Takes a very long time to load and shows a very basic HTML page with a refresh icon and non working search bar. I am sure the port is forwarded correctly. 

**Screenshots**
![Screenshot 2024-01-21 201748](https://github.com/Minecraft-Transit-Railway/Minecraft-Transit-Railway/assets/94990691/f8f1ba0f-bd84-4a55-8084-841e9c335d17)

**Environment**

- Minecraft Version: 1.16.5 Mohist server, running on a Raspberry Pi 4 8G (Runs quite well actually if your running a small community server)
- Installed Mods:

- Architectury 
- Create
- Flywheel
- Joy of Painting
- Lightman's Currency
- MTR
- More Minecarts
- Mr Crayfish's Furniture Mod
- Mr Crayfish's Gun Mod
- Obfuscate
- Plushie Mod
- Securitycraft
- XL Food Mod

**Comment by schooldanlp6** [comment]
yes I sadly cannot open system map at all

**Comment by Simmotipo** [comment]
Not sure if when you say should be fixed in 4.0.0, you mean the full release, but in 4.0.0-pre3 at least the issue still exists. Either way, putting these observations for anyone who may come across this thread in future--

I see in OP's screenshot they're accessing `http://HOST:8888/index.html` -- I'm personally not sure whether this is the expected URL to use, or simply `http://HOST:8888` (one that perhaps someone who has used the mod longer than I have could confirm), as the issue appears to be that when you access `/index.html`, incorrect URLs are used to load the API data.

Error in console:

![Image](https://github.com/user-attachments/assets/9ccbbe2a-72fc-4bc4-aac8-8ca813ac64af)

Having a skim over the highlighted line, and there are URLs formed via, for example `${document.location.origin}${document.location.pathname}mtr/api/map/${o}?dimension=${this.dimensionService.getDimensionIndex()}` - which is fine when just going to `HOST:8888`, but yields a URL along the lines of "http://HOST:8888/index.htmlmtr/api/map....` if you go to the `index.html`

Sure enough, this is what is requested, and automatically 301'd back to the home page, causing HTML data to be returned, not the expected JSON-

![Image](https://github.com/user-attachments/assets/9d1c717a-6183-4f67-a6cf-1fc187427aef)

tl;dr -- dropping `index.html` from the URL (going just to `http://HOST:8888/`) works/fixes the issue and you get a web page, but I'm not sure if that's the intended approach :)
